plugins/scaffolder-backend/src/lib/catalog/CatalogEntityClient.ts

```diff
@@ -26,36 +26,32 @@ import {
  * A catalog client tailored for reading out entity data from the catalog.
  */
 export class CatalogEntityClient {
-  private readonly discovery: PluginEndpointDiscovery;
+  private readonly catalogClient: CatalogClient;
 
   constructor(options: { discovery: PluginEndpointDiscovery }) {
-    this.discovery = options.discovery;
+    this.catalogClient = new CatalogClient({
```

Comment:
should not construct things in constructors - make the `CatalogClient` an argument to the constructor instead, and if necessary, make a static `create` method that does the constructing and gluing together of things

Reply:
I suggested this :grin:

It's to avoid changes to the backend package for now, and wait with that until we perhaps do a wider `CatalogClient` refactor

Reply:
Ok ok.